lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -6885,65 +6830,140 @@ class DecomposeAtenTypeAsOp : public OpRewritePattern<AtenTypeAsOp> {
 };
 } // namespace
 
-// AtenIndexTensorOp
+// Torch ops related to indexing tensors, e.g., AtenIndexTensor, AtenIndexPut.
 namespace {
-// The goal of this pattern is to eliminate none index in aten.Index.Tensor's
-// `indices` param for the ease of various backend. The detailed steps are:
-//    1. reorder input tensor so that the non-none index appears at adjacent
-//    positions.
-//    2. manually generate index tensor with some ops like iota, to replace the
-//    none index in `indices`
-//    3. replace the old aten.Index.Tensor with a new
-//    aten.Index.Tensor_hacked_twin.
-class DecomposeAtenIndexTensorOp : public OpRewritePattern<AtenIndexTensorOp> {
-public:
-  using OpRewritePattern::OpRewritePattern;
 
-  // TODO: It might be better to use aten.view op instead of mulitple
-  // aten.unsqueeze. But currently, torch-to-linalg pass has limited support for
-  // view on dynamic shapes, such as [?] -> [?,1,1,1]. Using aten.view op will
-  // cause relevant e2e tests fail.
-  static FailureOr<Value>
-  unsqueezeTensorAtTrailingDim(Operation *op, PatternRewriter &rewriter,
-                               Value input, int count) {
-    Location loc = op->getLoc();
-    Value constMinusOne = rewriter.create<Torch::ConstantIntOp>(
-        loc, rewriter.getI64IntegerAttr(-1));
-    Value result = input;
-    while (count--) {
-      auto unsqzTensorInfo =
-          unsqueezeTensor(rewriter, op, result, /*dim=*/constMinusOne);
-      if (failed(unsqzTensorInfo)) {
-        return failure();
-      }
-
-      result = *unsqzTensorInfo;
+// TODO: It might be better to use aten.view op instead of mulitple
```

Comment:
I would change this from a todo and just say that "unsqueeze is more easily optimized than a generic view, and we prefer to enjoy ops with more structure than less in compositions".

View is a very bad op to be adding in a lowering pipeline because it represents a loss of information.